SceneManager配下とLibrary配下のリファクタリングと修正

Moto0116 opened the pull request:
**修正内容**
- プロジェクトの設定でマクロ再定義を無視するようにした(DirectX11の警告があまりにもうるさいので)
- Library配下はT-SENでやってあるコメントブロックを適用した。
- 仮置き関数などを消した
- enumをクラス内に収めた
- 初期化と解放部分の修正
- 変数名とコメントの修正
- InputDeviceManagerの比較がおかしかったので修正
- [pull #81]での指摘部分修正
- Messageboxで出力する文字列を修正

issue #170

Moto0116 commented:
作業完了しました。
確認お願いします。

Keitaro0408 commented on src/KingdomCraft/KingdomCraft/Library/InputDeviceManager/MouseDevice/MouseDevice.cpp:
```diff
@@ -6,22 +6,39 @@
-		m_DIMouseState.rgbButtons[i] = MOUSEBUTTON_OFF;
-		m_OldDIMouseState.rgbButtons[i] = MOUSEBUTTON_OFF;
+		m_DIMouseState.rgbButtons[i] = 0;
+		m_OldDIMouseState.rgbButtons[i] = 0;
```
なんでdefineじゃないようにしたん？

Moto0116 replied:
m_DIMouseState.rgbButtons[i] は整数型でMOUSEBUTTON_OFFはenumなんで、初期化としておかしいから修正しました。

Keitaro0408 replied:
理解しました

motolz committed:
[issue #170] SceneManager配下とLibrary配下のリファクタリングと修正

- プロジェクトの設定でマクロ再定義を無視するようにした(DirectX11の警告がうるさいので)
- Library配下はT-SENでやってあるコメントブロックを適用した。
- 仮置き関数などを消した
- enumをクラス内に収めた
- 初期化と解放部分の修正
- 変数名とコメントの修正
- InputDeviceManagerの比較がおかしかったので修正
- [pull #81]での指摘部分修正
- Messageboxで出力する文字列を修正

Keitaro0408 reviewed (approved):
確認しました。
コミットまとめお願いします

Moto0116 commented:
@Keitaro0408 コミットまとめたんで、合流お願いします。

Keitaro0408 commented:
マージしておきます

Moto0116 commented:
合流したので閉じる
